src/librustc_typeck/check/wfcheck.rs

```diff
@@ -653,8 +653,10 @@ fn error_380(ccx: &CrateCtxt, span: Span) {
 
 fn error_392<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>, span: Span, param_name: ast::Name)
                        -> DiagnosticBuilder<'tcx> {
-    struct_span_err!(ccx.tcx.sess, span, E0392,
-                     "parameter `{}` is never used", param_name)
+    let mut err = struct_span_err!(ccx.tcx.sess, span, E0392,
```

Comment:
@GuillaumeGomez Actually I tried it first but code like this turns mutable reference of DiagnosticBuilder because of `span_label` but return value of function is DiagnosticBuilder(not reference) so I couldn't did like that:
```rust
struct_span_err!(ccx.tcx.sess, span, E0392,
                     "parameter `{}` is never used", param_name)
                     .span_label(span, &format!("unused type parameter"))
```
I don't know  another way to do it without declaring variable so I did it like that. If there is a better way, I'm happy to do that.